From 46c58d11aaa559a762a5a1e78261d0edde9b526a Mon Sep 17 00:00:00 2001
From: Fred Fan <r01011@freescale.com>
Date: Tue, 1 Jul 2008 15:55:33 +0800
Subject: [PATCH] ENGR00082464 IMX35 enable SCC clock and fix div0 bug for clock out

1. set enable/disable function and register information to SCC clock
2. Change the code to calculate the divider of clk_out

Signed-off-by: Fred Fan <r01011@freescale.com>
---
 arch/arm/mach-mx35/clock.c |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-mx35/clock.c b/arch/arm/mach-mx35/clock.c
index c304883..b47ef57 100644
--- a/arch/arm/mach-mx35/clock.c
+++ b/arch/arm/mach-mx35/clock.c
@@ -1265,6 +1265,9 @@ static struct clk rtic_clk = {
 static struct clk scc_clk = {
 	.name = "scc_clk",
 	.parent = &ipg_clk,
+	.enable_reg = MXC_CCM_CGR2,
+	.enable_shift = MXC_CCM_CGR2_SCC_OFFSET,
+	.disable = _clk_disable,
 };
 
 static struct clk sdma_clk[] = {
@@ -1472,7 +1475,7 @@ static void _clk_cko1_recalc(struct clk *clk)
 	post = (reg & MXC_CCM_COSR_CLKOUT_PRODIV_MASK) >>
 	    MXC_CCM_COSR_CLKOUT_PRODIV_OFFSET;
 
-	factor = ((reg & MXC_CCM_COSR_CLKOUTDIV_1) != 0) << 1;
+	factor = 1 << ((reg & MXC_CCM_COSR_CLKOUTDIV_1) != 0);
 	clk->rate = clk->parent->rate / (factor * (pre + 1) * (post + 1));
 }
 
-- 
1.5.4.4

